src/go/rpk/pkg/config/schema.go

```diff
@@ -94,26 +98,29 @@ type ServerTLS struct {
+	AdditionalStartFlags     []string             `yaml:"additional_start_flags,omitempty" mapstructure:"additional_start_flags,omitempty" json:"additionalStartFlags"`
+	EnableUsageStats         bool                 `yaml:"enable_usage_stats" mapstructure:"enable_usage_stats" json:"enableUsageStats"`
+	TuneNetwork              bool                 `yaml:"tune_network" mapstructure:"tune_network" json:"tuneNetwork"`
+	TuneDiskScheduler        bool                 `yaml:"tune_disk_scheduler" mapstructure:"tune_disk_scheduler" json:"tuneDiskScheduler"`
+	TuneNomerges             bool                 `yaml:"tune_disk_nomerges" mapstructure:"tune_disk_nomerges" json:"tuneNomerges"`
+	TuneDiskWriteCache       bool                 `yaml:"tune_disk_write_cache" mapstructure:"tune_disk_write_cache" json:"tuneDiskWriteCache"`
+	TuneDiskIrq              bool                 `yaml:"tune_disk_irq" mapstructure:"tune_disk_irq" json:"tuneDiskIrq"`
+	TuneFstrim               bool                 `yaml:"tune_fstrim" mapstructure:"tune_fstrim" json:"tuneFstrim"`
+	TuneCpu                  bool                 `yaml:"tune_cpu" mapstructure:"tune_cpu" json:"tuneCpu"`
+	TuneAioEvents            bool                 `yaml:"tune_aio_events" mapstructure:"tune_aio_events" json:"tuneAioEvents"`
+	TuneClocksource          bool                 `yaml:"tune_clocksource" mapstructure:"tune_clocksource" json:"tuneClocksource"`
+	TuneSwappiness           bool                 `yaml:"tune_swappiness" mapstructure:"tune_swappiness" json:"tuneSwappiness"`
+	TuneTransparentHugePages bool                 `yaml:"tune_transparent_hugepages" mapstructure:"tune_transparent_hugepages" json:"tuneTransparentHugePages"`
+	EnableMemoryLocking      bool                 `yaml:"enable_memory_locking" mapstructure:"enable_memory_locking" json:"enableMemoryLocking"`
+	TuneCoredump             bool                 `yaml:"tune_coredump" mapstructure:"tune_coredump" json:"tuneCoredump"`
+	CoredumpDir              string               `yaml:"coredump_dir,omitempty" mapstructure:"coredump_dir,omitempty" json:"coredumpDir"`
+	WellKnownIo              string               `yaml:"well_known_io,omitempty" mapstructure:"well_known_io,omitempty" json:"wellKnownIo"`
+	Overprovisioned          bool                 `yaml:"overprovisioned" mapstructure:"overprovisioned" json:"overprovisioned"`
+	SMP                      *int                 `yaml:"smp,omitempty" mapstructure:"smp,omitempty" json:"smp,omitempty"`
+	SCRAMUser                string               `yaml:"scram_user,omitempty" mapstructure:"scram_user,omitempty" json:"scram_user,omitempty"`
+	SCRAMPassword            string               `yaml:"scram_password,omitempty" mapstructure:"scram_password,omitempty" json:"scram_password,omitempty"`
+	SCRAMType                sarama.SASLMechanism `yaml:"scram_type,omitempty" mapstructure:"scram_type,omitempty" json:"scram_type,omitempty"`
```

Comment:
Please create a struct to hold SCRAM config, like we did with TLS, to prevent the `rpk` top-level config from getting more polluted. It can be something like
```go
type SCRAM struct {
    User string
    Password string
    Type string
}
```

Also, it seems `SASLMechanism` is just an alias for `string` https://github.com/Shopify/sarama/blob/65f0fec86aabe011db77ad641d31fddf14f3ca41/broker.go#L57 . I think we should avoid importing types we don't need.

Reply:
Ok changed.